storage-proofs/src/piece_inclusion_proof.rs

```diff
@@ -485,4 +566,84 @@ mod tests {
         assert_eq!(subtree_capacity(14, 16), 2);
         assert_eq!(subtree_capacity(15, 16), 1);
     }
+
+    fn test_pip_casting<H: Hasher>() {
+        let valid_proof_element_bytes = [
+            1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9,
+            0, 1, 2,
+        ];
+        let valid_proof_element = H::Domain::try_from_bytes(&valid_proof_element_bytes).unwrap();
+
+        let empty_pip: PieceInclusionProof<H> = PieceInclusionProof {
```

Comment:
TODO: Split the empty PIP serialization/deserialization roundtrip into a separate test.